README.md

```diff
@@ -1,16 +1,715 @@
+
+Receipts don't have their own version field. Receipts MUST use the same version as the invocation that they contain.
+
+## 4.1 Fields
+
+| Field  | Type            | Description                                                           | Required | Default |
+|--------|-----------------|-----------------------------------------------------------------------|----------|---------|
+| `inv`  | `&Invocation`   | CID of the Invocation that generated this response                    | Yes      |         |
+| `out`  | `{String: Any}` | The results of each call, the task's label. MAY contain sub-receipts. | Yes      |         |
+| `meta` | `Any`           | Non-normative extended fields                                         | No       | `null`  |
+
+### 4.1.1 Invocation
+
+The `inv` field MUST include a link to the Invocation that the Receipt is for.
+
+### 4.1.2 Output
+
+The `out` field MUST contain the output of steps of the call graph, indexed by the task name inside the invocation. If the invocation is the implicit `"*"`, then the base64 hash of the concatenation of the URI, Ability and extensional fields MUST be used.
+
+The `out` field MAY omit any tasks that have not yet completed, or results which are not public. An `Invocation` may be associated to zero or more `Receipts`.
+
+A `Result` MAY include recursive `Receipt` CIDs in on the `Success` branch. As a Task may require subdelegation, the OPTIONAL `rec` field MAY be used to include recursive `Receipt`s.
+
+### 4.1.3 Metadata Fields
+
+The metadata field MAY be omitted or used to contain additional data about the receipt. This field MAY be used for tags, commentary, trace information, and so on.
+
+## 4.2 IPLD Schema
+
+``` ipldsch
+type SignedReceipt struct {
+  rec Receipt (rename "ucan/receipt")
+  sig VarSig
+}
+
+type Receipt struct {
+  inv  &SignedInvocation
+  out  {String : Result}
+  meta optional Any
+} 
+
+type Result union {
+  | Success
+  | Failure
+}
+
+type Failure struct {
+  err   nullable String
+  trace Any
+}
+
+type Success struct {
+  val Any
+  rec optional &SignedReceipt
+}
+```
+
+## 4.3 JSON Example
+
+``` json
+{
+  "ucan/receipt": {
+    "v": "0.1.0",
+    "inv": [{"/": "bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"}],
+    "out": {
+      "ok": {
+        "bafkreiakkqwuffzsxrzseo7fweeicqlnqanhvyffjieh3etsbnnkjxbphi": [
+          {
+            "from": "alice@example.com",
+            "text": "Hello world!"
+          }, 
+          {
+            "from": "bob@example.com",
+            "text": "What's up?"
+          }
+        ],
+        "bafkreienxymjwglxlb3rkdyeyjt54ddoe4x4qi7a7hyfce3z56zspmy6pm": {
+          "http": { 
+            "status": 200,
+            "body": "hello world"
+          },
+          "ms": 476
+        },
+        "delegated-task": {"/": "bafkreieiupg4smeb5ammpsydbiea4yvwzwne5ly4hiripy4cjocqiat3ce"}
```

Comment:
this very specific field seems to make it seem like it should be part of possible `out` fields, no?

Reply:
Isn't it? 👀

Reply:
Oh actually I may need to explain that this is a link to a sub reference. I'm noticing a theme of "the examples could be make more clear and/or have more extensive ones in a separate directory"

Reply:
DISCLAIMER: yes I'm tagging Fission staff on the weekend, _BUT CTO LADY SAYS IGNORE UNTIL MONDAY!_

Related but directly on the above and maybe what you were trying to get at yesterday @zeeshanlakhani I wonder if we could let a downstream task take specifically the `stdout` or `stderr` sides. We can then further restrict that in IPVM, but this is a more general spec. This does mean differentiating between success and failure explicitly, or having an IPLD implicit 🤔

Reply:
So, I'm iffy on specifying `stdout` or `stderr`, as there can be `dev/null`. Now, having to specify success and failure options explicitly make fit better w/ the concept of promises around all these things?

Reply:
Ah, sorry, I should speak to you as my actual audience (you don't need the Unix analogy): the `Right` and `Left` sides are (as of my latest merge of the side PR) described as follows:

```json
{"ucan/ok": "some success value"}
{"ucan/err": "some failure value"}
```

Does that feel cleaner @zeeshanlakhani?

Reply:
Sure does @expede.

Reply:
and yep, now see why I didn't need the UNIX analogy haha.